File: spectra-serve-gateway/src/main/java/com/yangxj96/spectra/server/gateway/launch/AfterStartup.java
package com.yangxj96.spectra.server.gateway.launch;

import com.yangxj96.spectra.server.gateway.service.RouteService;
import jakarta.annotation.Resource;
import org.springframework.boot.CommandLineRunner;
import org.springframework.stereotype.Component;


/**
 * 启动完成后操作
 */
@Component
public class AfterStartup implements CommandLineRunner {

    @Resource
    private RouteService routeService;

    @Override
    public void run(String... args) throws Exception {
        routeService.refresh();
    }
}

File: yangxj96-serve/yangxj96-serve-gateway/src/main/java/com/yangxj96/saas/server/gateway/launch/AfterStartup.kt
package com.yangxj96.saas.server.gateway.launch

import com.yangxj96.saas.server.gateway.service.RouteService
import jakarta.annotation.Resource
import org.springframework.boot.CommandLineRunner
import org.springframework.stereotype.Component


/**
 * 启动完成后操作
 */
@Component
class AfterStartup : CommandLineRunner {

    @Resource
    private lateinit var routeService: RouteService

    override fun run(vararg args: String?) {
        routeService.refresh()
    }
}